app/services/api/settingsApi.ts

```diff
@@ -16,4 +16,18 @@ export const settingsApi = {
       error: (response.data as any)?.error || "Failed to retrieve dashboard preferences",
     } as ApiResponse<DashboardPreferences>
   },
+  setDashboardPreferences: async (preferences: DashboardPreferences): Promise<ApiResponse<any>> => {
```

Comment:
I think we can type this

```suggestion
setDashboardPreferences: async (preferences: DashboardPreferences): Promise<ApiResponse<DashboardPreferences>> => {
```